extension/src/test/XliffFunctions.test.ts

```diff
@@ -0,0 +1,84 @@
+import * as assert from "assert";
+import { TranslationMode } from "../Enums";
+import {
+  TargetState,
+  TranslationToken,
+  TransUnit,
+} from "../Xliff/XLIFFDocument";
+import * as XliffFunctions from "../XliffFunctions";
+
+suite("XliffFunctions Tests", function () {
+  test("formatTransUnitForTranslationMode - DTS", function () {
+    // DTS
+    const tu = getTransUnit(TranslationToken.notTranslated);
+    XliffFunctions.formatTransUnitForTranslationMode(TranslationMode.dts, tu);
+    assert.strictEqual(
+      tu.target.state,
+      TargetState.needsTranslation,
+      `Expected state "${TargetState.needsTranslation}".`
+    );
+    assert.strictEqual(tu.target.stateQualifier, undefined);
+    assert.strictEqual(tu.target.translationToken, undefined);
+  });
+
+  test("formatTransUnitForTranslationMode - External", function () {
+    // External
+    const tu = getTransUnit(TranslationToken.notTranslated);
+    XliffFunctions.formatTransUnitForTranslationMode(
+      TranslationMode.external,
+      tu
+    );
+    assert.strictEqual(
+      tu.target.state,
+      TargetState.needsTranslation,
+      `Expected state "${TargetState.needsTranslation}".`
+    );
+    assert.strictEqual(tu.target.stateQualifier, undefined);
+    assert.strictEqual(tu.target.translationToken, undefined);
+  });
+
+  test("formatTransUnitForTranslationMode - NAB Tags (default case)", function () {
+    // NAB Tags
+    const translationMode = TranslationMode.nabTags;
+    let tu = getTransUnit(undefined, TargetState.new);
+    XliffFunctions.formatTransUnitForTranslationMode(translationMode, tu);
+    assert.strictEqual(tu.target.state, undefined);
+    assert.strictEqual(tu.target.stateQualifier, undefined);
+    assert.strictEqual(
+      tu.target.translationToken,
+      TranslationToken.notTranslated
+    );
+
+    tu = getTransUnit(undefined, TargetState.needsReviewTranslation);
+    XliffFunctions.formatTransUnitForTranslationMode(translationMode, tu);
+    assert.strictEqual(tu.target.state, undefined);
+    assert.strictEqual(tu.target.stateQualifier, undefined);
+    assert.strictEqual(tu.target.translationToken, TranslationToken.review);
+
+    // Test default case of inner switch
+    tu = getTransUnit(undefined, TargetState.final);
+    XliffFunctions.formatTransUnitForTranslationMode(translationMode, tu);
+    assert.strictEqual(tu.target.state, undefined);
+    assert.strictEqual(tu.target.stateQualifier, undefined);
+    assert.strictEqual(tu.target.translationToken, undefined);
+  });
+});
+
+function getTransUnit(
```

Comment:
I rather split this into two functions, since both parameter should not be defined at the same time.. 
Eg.
getTransUnitWithState
getTransUnitWithToken

And if you remove the state from the `getTransUnitXml()` function, you don't need to set them to `undefined`

Reply:
Same low brain cell count to blame here. Latest commit solves this better.